python/cudf/cudf/core/column/column.py

```diff
@@ -1322,10 +1322,17 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
 
     elif np.isscalar(arbitrary) and not isinstance(arbitrary, memoryview):
         length = length or 1
+        if (
+            (nan_as_null is True)
+            and isinstance(arbitrary, float)
```

Comment:
This check will return `False` if you give a numpy float scalar, i.e. `np.dtype('float32').type(np.nan)`. Could we just remove it and only use the `np.isnan` check?

Reply:
In that case I need to check for `str` as isnan will error in that case.